libs/langchain/langchain/llms/huggingface_pipeline.py

```diff
@@ -3,18 +3,20 @@
 from typing import Any, List, Mapping, Optional
 
 from langchain.callbacks.manager import CallbackManagerForLLMRun
-from langchain.llms.base import LLM
+from langchain.llms.base import BaseLLM
 from langchain.llms.utils import enforce_stop_tokens
 from langchain.pydantic_v1 import Extra
+from langchain.schema import Generation, LLMResult
 
 DEFAULT_MODEL_ID = "gpt2"
 DEFAULT_TASK = "text-generation"
 VALID_TASKS = ("text2text-generation", "text-generation", "summarization")
+DEFAULT_BATCH_SIZE = 1
```

Comment:
Why default to 1?

Reply:
The idea was backwards compat... not to surprise people with OOM if they already have a pipeline running without batching... I believe the way the code is written it would do batching even if they called .invoke(...) not .batch(...)

Reply:
... or perhaps this is my lack of understanding of the new calling pattern of generate... and perhaps in the case of .invoke(...) generate gets called one prompt at a time anyway? In that case yes we should be able to set batch size to a higher number (I think 4 might be appropriate though it really depends on model size and GPU size)

Let me know, I can make a quick fix to change this to 4... thanks!

Reply:
For people using .invoke() it would call this with an array with a single prompt, so I imagine memory usage would be same as before? If that's your understanding too I'd be happy to default to 4

Reply:
Done :)